### PR TITLE
Create CoreData: Create Enity, Attribute and CoreDataManager

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 73907: Create CoreData](https://edu-redmine.sun-asterisk.vn/issues/73907)
## WHAT
- Create CoreData: Entity, Attribute
- Create CoreDataManager
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/Zts4P8 |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Message | Sun* commit message | Template: [TASK/FREATURE/BUG/SUPPORT][RedmineID] + message |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: Weather_App/Weather_App/CoreData/CoreDataManager.swift [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+//  CoreDataManager.swift
+//  Weather_App
+//
+//  Created by ho.bao.an on 25/03/2024.
+//
+
+import Foundation
+import CoreData
+
+class CoreDataManager {
```
Comment (ghost) [inline Weather_App/Weather_App/CoreData/CoreDataManager.swift]: final

File: Weather_App/Weather_App/CoreData/CoreDataManager.swift [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            do {
+                try context.save()
+            } catch {
+                let nserror = error as NSError
+                fatalError("Core Data: Unresolved error \(nserror), \(nserror.userInfo)")
+            }
+        }
+    }
+    
+    func fetchData() -> [WeatherEntity] {
```
Comment (ghost) [inline Weather_App/Weather_App/CoreData/CoreDataManager.swift]: dùng generic để mình tái sử dụng nhiều hơn e k nên fix cứng loại type cho nó

File: Weather_App/Weather_App/CoreData/CoreDataManager.swift [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        let fetchRequest: NSFetchRequest<WeatherEntity> = WeatherEntity.fetchRequest()
+        do {
+            return try context.fetch(fetchRequest)
+        } catch {
+            print("Core Data: Fetch failed: \(error.localizedDescription)")
+            return []
+        }
+    }
+    
+    func addData(latitude: Double, longitude: Double, nameCity: String,saveStaus: Bool) {
```
Comment (ghost) [inline Weather_App/Weather_App/CoreData/CoreDataManager.swift]: tương tự làm cho nó tái sử dụng được k chỉ riêng model weather

File: Weather_App/Weather_App/CoreData/CoreDataManager.swift [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            newWeatherEntity.latitude = latitude
+            newWeatherEntity.longitude = longitude
+            newWeatherEntity.nameCity = nameCity
+            newWeatherEntity.saveStatus = saveStaus
+            saveContext()
+            print("Core Data: Save data success")
+        }
+    }
+    
+    func deleteData(_ weatherEntity: WeatherEntity) {
```
Comment (ghost) [inline Weather_App/Weather_App/CoreData/CoreDataManager.swift]: này cũng vậy

**Review by daolq-2712 — APPROVED** [review]
OK!

**Pull request merged**